src/Raven.Server/Documents/ETL/Providers/AI/AiHelper.cs

```diff
@@ -0,0 +1,127 @@
+using Raven.Server.ServerWide;
+
+namespace Raven.Server.Documents.ETL.Providers.AI;
+
+public static class AiHelper
+{
+    private static readonly IBlake2B Hash;
+
+    static AiHelper()
+    {
+        Hash = Blake2BFactory.Instance.Create(new Blake2BConfig
+        {
+            HashSizeInBits = 256
+        });
+    }
+
+    public static string CalculateValueHash(string value)
+    {
+        return Hash.ComputeHash(value).AsHexString(uppercase: true);
+    }
+
+    public static string GetDocumentEmbeddingsId(string documentId)
+    {
+        return $"{documentId}/embeddings";
+    }
+
+    public static string GetDocumentEmbeddingsCollectionName(string sourceCollectionName)
+    {
+        return $"{sourceCollectionName}/embeddings";
+    }
+
+    public static string GetValueEmbeddingsDocumentId(string configurationName, string hash)
+    {
+        return $"embeddings/{configurationName}/{hash}";
+    }
+
+    [Experimental("SKEXP0001")]
+    public static ITextEmbeddingGenerationService CreateService(AiEtlConfiguration configuration)
+    {
+        var kernelBuilder = Kernel.CreateBuilder();
+
+        switch (configuration.AiConnectorType)
+        {
+            case AiConnectorType.OpenAi:
+                var openAiSettings = configuration.Connection.OpenAiSettings;
+
+                var apiKey = new ApiKeyCredential(openAiSettings.ApiKey);
+                var openAiOptions = new OpenAIClientOptions
+                {
+                    Endpoint = new Uri(openAiSettings.Endpoint),
+                    ProjectId = openAiSettings.ProjectId,
+                    UserAgentApplicationId = $"RavenDB/{ServerVersion.Version}/{nameof(AiEtl)}"
+                };
+                var openAIClient = new OpenAIClient(apiKey, openAiOptions);
+                kernelBuilder.AddOpenAITextEmbeddingGeneration(openAiSettings.Model, openAIClient);
+
+                break;
+
+            case AiConnectorType.AzureOpenAI:
+                var azureOpenAiSettings = configuration.Connection.AzureOpenAiSettings;
+
+                kernelBuilder.AddAzureOpenAITextEmbeddingGeneration(
+                    azureOpenAiSettings.DeploymentName,
+                    azureOpenAiSettings.Endpoint,
+                    azureOpenAiSettings.ApiKey,
+                    azureOpenAiSettings.Model);
+
+                break;
+
+            case AiConnectorType.Ollama:
+                var ollamaSettings = configuration.Connection.OllamaSettings;
+                var ollamaApiConfig = new OllamaApiClient.Configuration
+                {
+                    Uri = new Uri(ollamaSettings.Uri),
+                    Model = ollamaSettings.Model
+                };
+
+                var ollamaApiClient = new OllamaApiClient(ollamaApiConfig);
+
+                kernelBuilder.AddOllamaTextEmbeddingGeneration(ollamaApiClient);
+
+                // var modelInfo = AsyncHelpers.RunSync(() => ollamaApiClient.ShowModelAsync(ollamaSettings.Model));
```

Comment:
do we need this?

Reply:
I'll delete it on the step of the implementation of connection string tests if you don't mind.